openstudiocore/src/model/AirLoopHVACOutdoorAirSystem.cpp

```diff
@@ -633,6 +637,28 @@ namespace detail {
     return false;
   }
 
+  boost::optional<AirflowNetworkDistributionNode> AirLoopHVACOutdoorAirSystem_Impl::createAirflowNetworkDistributionNode()
+  {
+    boost::optional<AirflowNetworkDistributionNode> opt = airflowNetworkDistributionNode();
```

Comment:
This is a bit backwards of how I see this.  I don't think people really care if this got created or not (if they did they could call airflowNetworkDistributionNode first and see if it returns empty).  I would make this:

AirflowNetworkDistributionNode AirLoopHVACOutdoorAirSystem_Impl::getOrCreateAirflowNetworkDistributionNode()

Reply:
I also like the name demandAirflowNetworkDistributionNode but I think demand could cause confusion about supply/demand side stuff

Reply:
Also, AirflowNetworkDistributionNode getAirflowNetworkDistributionNode is not too bad

Reply:
Now that @joseph-robertson 's AdditionalProperties object has been merged, i think that is the API to reference:

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/ModelObject.hpp#L278

    airflowNetworkDistributionNode // create or get the node
    hasAirflowNetworkDistributionNode // check if the node already exists
    removeAirflowNetworkDistributionNode // convenience, equivalent to if (hasAirflowNetworkDistributionNode){airflowNetworkDistributionNode.remove}

Reply:
I realize this is potentially a big change to ask for, but also feel like it is right to get the API right the first time.  @kbenne can you weigh in here?

Reply:
I see this same implementation in many (all?) of the other classes. can it be pushed down the inheritance to a common location? or factored into a common function that is called from the implementation here?

Also passing handle into the constructor seems unusual to me. Seems like a new pattern. 

The createAFNNode method itself is also new to me. Perhaps that is what Dan is getting at. I do prefer the get or create as needed behavior of :: airflowNetworkDistributionNode.